src/plugins/pcr_primer_design_for_dna_assembly/src/PCRPrimerDesignForDNAAssemblyOPWidget.cpp

```diff
@@ -0,0 +1,249 @@
+    if (smButton == tbLeftAreaSelectManually) {
+        sbStartRegion = sbLeftAreaStart;
+        sbEndRegion = sbLeftAreaEnd;
+    } else {
+        sbStartRegion = sbRightAreaStart;
+        sbEndRegion = sbRightAreaEnd;
+    }
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
+
+    auto sequenceSelection = activeSequenceContext->getSequenceSelection();
+    SAFE_POINT(sequenceSelection != nullptr, L10N::nullPointerError("DNASequenceSelection"), );
+
+    updateRegionConnection = connect(sequenceSelection, &DNASequenceSelection::si_onSelectionChanged, this, &PCRPrimerDesignForDNAAssemblyOPWidget::sl_updateRegion);
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_updateRegion() {
+    auto sequenceSelection = qobject_cast<DNASequenceSelection*>(sender());
+    SAFE_POINT(sequenceSelection != nullptr, L10N::nullPointerError("DNASequenceSelection"), );
+
+    const auto& selectedRegions = sequenceSelection->getSelectedRegions();
+    CHECK(!selectedRegions.isEmpty(), );
+
+    if (selectedRegions.size() > 1) {
+        coreLog.details(tr("Multiple regions selection, the only first is set as primer search area."));
+    }
+    const auto& region = selectedRegions.first();
+    SAFE_POINT(sbStartRegion != nullptr, L10N::nullPointerError("QSpinBox"), );
+
+    sbStartRegion->setValue(region.startPos + 1);
+    SAFE_POINT(sbEndRegion != nullptr, L10N::nullPointerError("QSpinBox"), );
+
+    sbEndRegion->setValue(region.endPos());
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_updateMaxValues() {
+    auto activeSequenceContext = annDnaView->getActiveSequenceContext();
+    SAFE_POINT(activeSequenceContext != nullptr, L10N::nullPointerError("ADVSequenceObjectContext"), );
+
+    auto seqLength = activeSequenceContext->getSequenceLength();
+    const auto& parametersMinMaxSpinBoxesKeys = parametersMinMaxSpinBoxes.keys();
+    for (const auto& minSb : qAsConst(parametersMinMaxSpinBoxesKeys)) {
+        auto maxSb = parametersMinMaxSpinBoxes.value(minSb);
+        SAFE_POINT(maxSb != nullptr, L10N::nullPointerError("QSpinBox"), );
+
+        minSb->setMaximum(maxSb->value() - 1);
+        maxSb->setMinimum(minSb->value() + 1);
+        maxSb->setMaximum(seqLength);
+    }
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_minValueChanged() {
+    auto minSb = qobject_cast<QSpinBox*>(sender());
+    SAFE_POINT(minSb != nullptr, L10N::nullPointerError("QSpinBox"), );
+
+    auto maxSb = parametersMinMaxSpinBoxes.value(minSb);
+    SAFE_POINT(maxSb != nullptr, L10N::nullPointerError("QSpinBox"), );
+
+    maxSb->setMinimum(minSb->value() + 1);
+}
+
+void PCRPrimerDesignForDNAAssemblyOPWidget::sl_maxValueChanged() {
+    auto maxSb = qobject_cast<QSpinBox*>(sender());
+    SAFE_POINT(maxSb != nullptr, L10N::nullPointerError("QSpinBox"), );
+
+    auto minSb = parametersMinMaxSpinBoxes.key(maxSb);
```

Comment:
Doc for  'parametersMinMaxSpinBoxes' says: "minimum values as keys and spinboxes with maximum values as values".
Is it safe to use maxSb as a key?

Reply:
We use here `maxSb` as value. See [here](https://doc.qt.io/qt-5/qmap.html#key)

Reply:
Ok, got it.